descartes_light/test/utest.cpp

```diff
@@ -225,6 +294,75 @@ TYPED_TEST(SolverFixture, KnownPathTest)
   }
 }
 
+TYPED_TEST(ParameterizedSolverFixture, ParameterizedKnownPath)
+{
+  using FloatType = typename TypeParam::FloatType;
+
+  // Build a graph where one sample for each waypoint is an all zero state; evaluate edges using the Euclidean distance
+  // metric Since each waypoint has an all-zero state, the shortest path should be through these samples
+  auto edge_eval = std::make_shared<const EuclideanDistanceEdgeEvaluator<FloatType>>();
+  auto state_eval = std::make_shared<const NaiveStateEvaluator<FloatType>>(true, this->state_cost);
+
+  std::size_t dof_min = 4;
+  std::size_t dof_max = 8;
+  std::size_t n_waypoints_min = 12;
+  std::size_t n_waypoints_max = 20;
+  std::size_t samples_per_waypoint_min = 5;
+  std::size_t samples_per_waypoint_max = 10;
+
+  for (std::size_t dof = dof_min; dof <= dof_max; dof++)
+  {
+    for (std::size_t n_waypoints = n_waypoints_min; n_waypoints <= n_waypoints_max; n_waypoints++)
+    {
+      for (std::size_t samples_per_wp = samples_per_waypoint_min; samples_per_wp <= samples_per_waypoint_max;
+           samples_per_wp++)
+      {
+        std::vector<typename WaypointSampler<FloatType>::ConstPtr> samplers =
+            createSamplers<FloatType>(dof, n_waypoints, samples_per_wp, this->state_cost);
+
+        typename Solver<FloatType>::Ptr solver = this->configurator.create();
+
+        // building graph
+        PerformanceResults<BuildStatus> build_performance_results = measurePerformance<BuildStatus>(
+            [&]() -> BuildStatus { return solver->build(samplers, { edge_eval }, { state_eval }); });
+        BuildStatus status = build_performance_results.return_val;
```

Comment:
The `PerformanceResults` struct and `measurePerformance` function seem overly complicated. Let's just measure the time around the two functions we care about. We can revise this later when we add more functionality if needed. I imagine the measurement of memory usage would be something totally separate anyway

```suggestion
        auto start_time = steady_clock::now();
        BuildStatus status = solver->build(samplers, { edge_eval }, { state_eval });
        auto build_time = (steady_clock::now() - start_time).count();
        ...
        start_time = steady_clock::now();
        SearchResult<FloatType> result = solver->search();
        auto search_time = (steady_clock::now() - start_time).count();
        ...
        std::cout << "Build time: " << build_time << "\nSearch time: " << search_time << std::endl;
```

Reply:
Why not use google benchmarks for this? You could also then track this over time for every commit and have a graphical visualization like what you see [here](https://ros-industrial-consortium.github.io/tesseract/dev/bench/)

Reply:
I intended for measurePerformance function to be something that we would extend as we added more metrics that we would want to measure.  That way we could make all these measurements by calling a single function rather than copying and pasting a code snippet that computes/prints the metrics and tweaking it to match the local variables.  However I can change to what you suggested if you're ok with the snippet copy paste approach.

Reply:
The benchmark was moved to a benchmark directory where we plan to use Google Benchmarks in the future. The goal is to eventually show the performance of the different solver implementations relative to one another for various sizes of problems.